src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1269,21 +1272,88 @@ void OrbitMainWindow::ShowTooltip(std::string_view message) {
                      QString::fromUtf8(message.data(), static_cast<int>(message.size())), this);
 }
 
+std::optional<QString> OrbitMainWindow::LoadSourceCode(const std::filesystem::path& file_path) {
+  QFile source_code_file{QString::fromStdString(file_path.string())};
+
+  if (source_code_file.open(QIODevice::ReadOnly)) return source_code_file.readAll();
+
+  orbit_source_paths_mapping::MappingManager mapping_manager{};
+  const auto maybe_mapping_file_path = mapping_manager.MapToFirstExistingTarget(file_path);
+  if (maybe_mapping_file_path.has_value()) {
+    source_code_file.setFileName(QString::fromStdString(maybe_mapping_file_path->string()));
+    if (source_code_file.open(QIODevice::ReadOnly)) return source_code_file.readAll();
+  }
+
+  QMessageBox message_box{QMessageBox::Warning, "Source code file not found",
+                          QString("Could not find the source code file \"%1\" on this machine.")
+                              .arg(QString::fromStdString(file_path.string())),
+                          QMessageBox::Cancel, this};
+  auto* pick_file_button = message_box.addButton("Choose file...", QMessageBox::ActionRole);
+
+  constexpr const char* kAutocreateMappingKey = "auto_create_mapping";
+  constexpr const char* kPreviousSourcePathsMappingDirectoryKey =
+      "previous_source_paths_mapping_directory";
+  auto checkbox = std::make_unique<QCheckBox>(
+      "Automatically create a source paths mapping from my chosen file.");
+  checkbox->setToolTip(
+      "If enabled Orbit will automatically try to create a source paths mapping from it. The "
+      "common suffix between the path given in the debug information and the local file path will "
+      "be stripped. From the rest a mapping will be created.");
+  QSettings settings{};
+  checkbox->setCheckState(settings.value(kAutocreateMappingKey, true).toBool() ? Qt::Checked
+                                                                               : Qt::Unchecked);
+  message_box.setCheckBox(checkbox.release());
```

Comment:
This line took me a minute to understand. A comment about transferring of ownership would be helpful

Reply:
Done.